packages/calcite-components/src/components/input-number/input-number.e2e.ts

```diff
@@ -1662,6 +1662,34 @@ describe("calcite-input-number", () => {
     expect(calciteInputNumberInput).toHaveReceivedEventTimes(1);
   });
 
+  it("emits change event when value set directly and then cleared in 'de' locale", async () => {
+    const page = await newE2EPage();
+    await page.setContent(`
+      <calcite-input-number lang="de" value="0" clearable></calcite-input-number>
+    `);
+
+    const calciteInputNumberChange = await page.spyOnEvent("calciteInputNumberChange");
+    const inputEl = await page.find("calcite-input-number");
+    const clearButtonEl = await page.find("calcite-input-number >>> .clear-button");
+
+    await page.evaluate(async () => {
```

Comment:
From what I understand, `setContent` will resolve after all components have fired `componentOnReady`. Could you test to confirm this is needed? I think you could simplify to the following:

```tsx
await waitForFrame(); // might not be needed if `setContent` resolution also includes `lang` updates
await inputEl.setProperty("value", "49.173126");
```